Content.Server/Chemistry/Components/MeleeChemicalPickerComponent.cs

```diff
@@ -0,0 +1,25 @@
+using Content.Shared.FixedPoint;
+
+namespace Content.Server.Chemistry.Components
+{
+    [RegisterComponent]
+    public sealed partial class MeleeChemicalPickerComponent : Component
+    {
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("transferAmount")]
+        public FixedPoint2 TransferAmount { get; set; } = FixedPoint2.New(1);
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float TransferEfficiency { get => _transferEfficiency; set => _transferEfficiency = Math.Clamp(value, 0, 1); }
+
+        [DataField("transferEfficiency")]
+        private float _transferEfficiency = 1f;
```

Comment:
this isn't really a pattern used anymore. Just make a single variable and add a comment specifying that it's meant to represent a percentage. If someone wants, they should be able to make it exceed 100%.